src/kOS.Safe/Encapsulation/PrimitiveStructure.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System;
+
+namespace kOS.Safe.Encapsulation
+{
+    [kOS.Safe.Utilities.KOSNomenclature("Structure", KOSToCSharp = false)]
+    public abstract class PrimitiveStructure : Structure
+    {
+        public abstract object ToPrimitive();
+    }
+}
+
```

Comment:
It looks to me like you're replacing the `IserializableValue` interface with the `PrimitiveStructure` inherited class.  Since this class only serves to force the object to implement `ToPrimitive` is there a reason you didn't add the method to the previous interface?